dags/sentinel-2_data_transfer.py

```diff
@@ -25,24 +23,48 @@
 from airflow.hooks.S3_hook import S3Hook
 
 default_args = {
-    'owner': 'Airflow',
+    "owner": "Airflow",
     "start_date": datetime(2020, 6, 12),
-    'email': ['toktam.ebadi@ga.gov.au'],
-    'email_on_failure': True,
-    'email_on_retry': False,
-    'retries': 0,
-    'num_workers': 10,
-    'africa_tiles': "data/africa-mgrs-tiles.csv",
-    'africa_conn_id': "deafrica-prod-migration",
+    "email": ["toktam.ebadi@ga.gov.au"],
+    "email_on_failure": True,
+    "email_on_retry": False,
+    "retries": 0,
+    "num_workers": 10,
+    "africa_conn_id": "deafrica-prod-migration",
     "us_conn_id": "deafrica-migration_us",
     "dest_bucket_name": "deafrica-sentinel-2",
     "src_bucket_name": "sentinel-cogs",
     "schedule_interval": "0 */1 * * *",
     "sentinel2_topic_arn": "arn:aws:sns:af-south-1:543785577597:deafrica-sentinel-2-scene-topic",
-    "sqs_queue": ("https://sqs.us-west-2.amazonaws.com/565417506782/"
-                  "deafrica-prod-eks-sentinel-2-data-transfer")
+    "sqs_queue": "deafrica-prod-eks-sentinel-2-data-transfer",
 }
 
+
+def get_messages(
+    queue,
+    limit: bool = None,
+    visibility_timeout: int = 60,
+    message_attributes: Iterable[str] = ["All"],
+):
+    """
+    Get messages from a queue resource.
+    """
+    count = 0
+    while True:
+        messages = queue.receive_messages(
+            VisibilityTimeout=visibility_timeout,
+            MaxNumberOfMessages=1,
+            WaitTimeSeconds=10,
+            MessageAttributeNames=message_attributes,
+        )
+        if len(messages) == 0 or (limit and count >= limit):
+            break
+        else:
+            for message in messages:
+                count += 1
+                yield message
+
+
 def extract_src_key(src_url):
```

Comment:
There's a function that returns the `bucket` and `key` from a `s3url`. It's called `parse_s3_url` and returns `bucket_name, key`

Reply:
Maybe this method is better, though... I just think it's an odd way to parse URIs, doing a regex for slashes!

Reply:
It's ugly, I've replaced it with parse_s3_url.  It was useful when I had https links but that gets converted to s3 path before it reaches here.